src/rust/bitbox02-sys/build.rs

```diff
@@ -0,0 +1,41 @@
+use std::process::Command;
+use std::fs::File;
+use std::io::prelude::*;
+use temp_dir::TempDir;
+
+fn main() {
+    let path_to_bindings = if let Ok(cmake_dir) = std::env::var("CMAKE_CURRENT_BINARY_DIR") {
+        // if we are being invoked from CMAKE, the bindings are here:
+        format!("{}/rust", cmake_dir)
+    } else {
+        let bitbox02_sys_dir = std::env::var("CARGO_MANIFEST_DIR").unwrap();
+        let cmake_dir = format!("{}/../../../", bitbox02_sys_dir);
+
+        // generate list of includes using CMake
+        let tempdir = TempDir::with_prefix("bitbox02").unwrap();
+        let tempdir = tempdir.path().to_str().unwrap();
+        let _ = Command::new("cmake").arg(&cmake_dir).current_dir(&tempdir).output().unwrap();
+        let _ = Command::new("make").arg("rust-bindgen-includes").current_dir(&tempdir).output().unwrap();
+        let mut includes_file = File::open(format!("{}/src/rust-bindgen.flags", tempdir)).unwrap();
```

Comment:
I panic here if I go for example to `src/rust/bitbox02-rust/src/hww/api/set_device_name.rs` and run `cargo test`. Any idea?

Reply:
I'm not sure exactly what you see, but I see a thing that think is wrong. When you run "cargo test" on a crate it will not compile "test versions" of the dependencies. So basically the tests in "bitbox02-rust" should not depend on a feature called "testing" in the "bitbox02" crate.

Reply:
One workaround seems to be to use `cargo test --all-features`. But the "testing" feature isn't additive, so it doesn't compile.

https://doc.rust-lang.org/cargo/reference/features.html

> A consequence of this is that features should be additive. That is, enabling a feature should not disable functionality, and it should usually be safe to enable any combination of features.

Reply:
Good point, I forgot to enable the testing feature, but as you noticed, it still does not work.

Rust-analyzer seems to be working with this PR though, and also clippy seems to work. Just `cargo test` seems to fail at this line. The error is:

```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }', bitbox02-sys/build.rs:19:91
```

Reply:
Yeah cargo test is more complicated than clippy and RA since it has to do the final step of linking a working executable together. I'll see how hard it would be to support it